easybuild/framework/easyconfig/tools.py

```diff
@@ -138,23 +140,33 @@ def find_resolved_modules(easyconfigs, avail_modules, modtool, retain_all_deps=F
             elif retain_all_deps and dep_mod_name not in avail_modules:
                 # if all dependencies should be retained, include dep unless it has been already
                 _log.debug("Retaining new dep %s in 'retain all deps' mode", dep_mod_name)
-                deps.append(dep)
+                isresolved = False
 
             # retain dep if it is (still) in the list of easyconfigs
             elif dep_mod_name in ec_mod_names:
                 _log.debug("Dep %s is (still) in list of easyconfigs, retaining it", dep_mod_name)
-                deps.append(dep)
+                isresolved = False
 
             # retain dep if corresponding module is not available yet;
             # fallback to checking with modtool.exist is required,
             # for hidden modules and external modules where module name may be partial
             elif dep_mod_name not in avail_modules and not modtool.exist([dep_mod_name], skip_avail=True)[0]:
                 # no module available (yet) => retain dependency as one to be resolved
                 _log.debug("No module available for dep %s, retaining it", dep)
+                isresolved = False
+
+            if isresolved:
+                if 'full_mod_name' in dep:
+                    resolved_deps.append(dep['full_mod_name'])
```

Comment:
Shouldn't this be just (without the if-stmt)
```
resolved_deps.append(dep_mod_name)
```
Or to make the code behave the same way in both cases, just
```
resolved_deps.append(dep)
```